plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/type/TypeInfoUtils.java

```diff
@@ -0,0 +1,328 @@
+                    case BIGNUMERIC -> {
+                        if (params.length == 0) {
+                            yield getBigDecimalTypeInfo(76, 38);
+                        }
+                        if (params.length == 1) {
+                            yield getBigDecimalTypeInfo(parseInt(params[0]), 0);
+                        }
+                        if (params.length == 2) {
+                            yield getBigDecimalTypeInfo(parseInt(params[0]), parseInt(params[1]));
+                        }
+                        throw new IllegalArgumentException("Type decimal only takes two parameters, but %s is seen".formatted(params.length));
+                    }
+                    default -> getPrimitiveTypeInfo(typeEntry);
+                };
+            }
+
+            if (ARRAY.name().equals(token.text())) {
+                expect("<");
+                TypeInfo listElementType = parseType();
+                expect(">");
+                return getListTypeInfo(listElementType);
+            }
+
+            if (STRUCT.name().equals(token.text())) {
+                List<String> fieldNames = new ArrayList<>();
+                List<TypeInfo> fieldTypeInfos = new ArrayList<>();
+                boolean first = true;
+                do {
+                    if (first) {
+                        expect("<");
+                        first = false;
+                    }
+                    else {
+                        Token separator = expect(">", ",");
+                        if (separator.text().equals(">")) {
+                            break;
+                        }
+                        expect(" ");
+                    }
+                    Token name = expect("name", ">");
+                    if (name.text().equals(">")) {
+                        break;
+                    }
+                    fieldNames.add(name.text());
+                    expect(" ");
+                    fieldTypeInfos.add(parseType());
+                }
+                while (true);
+
+                return getStructTypeInfo(fieldNames, fieldTypeInfos);
+            }
+
+            throw new RuntimeException("Internal error parsing position %s of '%s'".formatted(token.position(), typeInfoString));
+        }
+
+        public PrimitiveParts parsePrimitiveParts()
+        {
+            Token token = expect("type");
+            return new PrimitiveParts(token.text(), parseParams());
+        }
+    }
+
+    public record PrimitiveParts(String typeName, String[] typeParams)
+    {
+        public PrimitiveParts
+        {
+            requireNonNull(typeName, "typeName is null");
+            requireNonNull(typeParams, "typeParams is null");
+        }
+    }
+
+    public static PrimitiveParts parsePrimitiveParts(String typeInfoString)
+    {
+        return new TypeInfoParser(typeInfoString).parsePrimitiveParts();
+    }
+
+    public static List<TypeInfo> getTypeInfosFromTypeString(String typeString)
```

Comment:
When do we get more than one entry from BigQuery information_schema.columns? I think we don't need this method and testMultipleTypes.